### PR TITLE
dra-cla: init at 2023-03-10

### PR DESCRIPTION
**Pull request by idlip**
###### Description of changes

 Package: dra-cla
 Source: https://github.com/CoolnsX/dra-cla
 Desc: Its an shell script cli tool to browse and play korean, chinese & asian drama. It is Similar to `ani-cli` package.
 
 I'm fairly still new to nix, this is my first time here. Hope so everything is right, as its a simple shell-script, it does `fetchgit` and moves the script file to path.
 The source git, does not have a tag or tar.gz file, so I have used `fetchgit` instead of `fetchFromGithub`, hope so this is right and serves the purpose!
 

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by idlip** [comment]
I have built this with these commands, and it worked as expected:

```
 cd pkgs/applications/video/dra-cla/
 nix-build -E 'with import <nixpkgs> {}; callPackage ./default.nix {}'
```

Or also worked when i added callPackage in `pkgs/top-level/all-packages.nix`
```
nix-build -A dra-cla
```

**Comment by idlip** [comment]
Sorry for too many commit, I squashed it to single commit! Hope so its clean

File: pkgs/applications/video/dra-cla/default.nix [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+    wrapProgram $out/bin/dra-cla \
+      --prefix PATH : ${lib.makeBinPath [ gnugrep gnused curl fzf mpv aria2 openssl ]}
+
+    runHook postInstall
+  '';
+
+  meta = with lib; {
+    homepage = "https://github.com/CoolnsX/dra-cla";
+    description = " A cli tool to browse and play korean drama, chinese drama Resources";
```
Comment (ocfox) [inline pkgs/applications/video/dra-cla/default.nix:39]:
```suggestion
    description = "A cli tool to browse and play korean drama, chinese drama Resources";
```

File: pkgs/applications/video/dra-cla/default.nix [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+, gnugrep
+, gnused
+, curl
+, fzf
+, mpv
+, aria2
+, openssl
+}:
+
+stdenvNoCC.mkDerivation rec {
```
Comment (ocfox) [inline pkgs/applications/video/dra-cla/default.nix:14]:
```suggestion
stdenvNoCC.mkDerivation {
```

**Review by ocfox — CHANGES_REQUESTED** [review]

**Comment by idlip** [comment]
@ocfox Thank you for reviewing this!

I have changed and commit as suggested, should i edit the commit message to "dra-cla: init at unstable-2023-03-10" ?

**Comment by idlip** [comment]
Actually there was no need for fzf, so removed that. (my bad)

File: pkgs/applications/video/dra-cla/default.nix [hunk 3]
```diff
@@ -0,0 +1,43 @@
+{ fetchgit
```
Comment (Janik-Haag) [inline pkgs/applications/video/dra-cla/default.nix:1]:
```suggestion
{ fetchFromGitHub
```

File: pkgs/applications/video/dra-cla/default.nix [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+, makeWrapper
+, stdenvNoCC
+, lib
+, gnugrep
+, gnused
+, curl
+, mpv
+, aria2
```
Comment (Janik-Haag) [inline pkgs/applications/video/dra-cla/default.nix:9]: The instructions of dra-cla states that ffmpeg is one of it's dependencys 
```suggestion
, aria2
, ffmpeg
```

File: pkgs/applications/video/dra-cla/default.nix [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+
+  nativeBuildInputs = [ makeWrapper ];
+
+  installPhase = ''
+      runHook preInstall
+
+    install -Dm755 dra-cla $out/bin/dra-cla
+
+    wrapProgram $out/bin/dra-cla \
+      --prefix PATH : ${lib.makeBinPath [ gnugrep gnused curl mpv aria2 openssl ]}
```
Comment (Janik-Haag) [inline pkgs/applications/video/dra-cla/default.nix:31]:
```suggestion
      --prefix PATH : ${lib.makeBinPath [ gnugrep gnused curl mpv aria2 ffmpeg openssl ]}
```

**Review by Janik-Haag — CHANGES_REQUESTED** [review]
Except for the fetchgit and the missing ffmpeg dependency it's fine I think.

**Comment by idlip** [comment]
@Janik-Haag Thank you for going through, I knew about ffmpeg and thought it would be dependency for mpv already, maybe ffmpeg is also used for downloading sometimes.

I have made the changed as suggested, the hash wasn't working for me to build, so updated it as nix threw my the hash value.

It works!

**Review by Janik-Haag — APPROVED** [review]

File: pkgs/applications/video/dra-cla/default.nix [hunk 6]
```diff
@@ -0,0 +1,45 @@
+{ fetchFromGitHub
+, makeWrapper
+, stdenvNoCC
+, lib
```
Comment (figsoda) [inline pkgs/applications/video/dra-cla/default.nix:4]: > [Arguments should be listed in the order they are used, with the exception of `lib`, which always goes first.](https://nixos.org/manual/nixpkgs/unstable/#sec-syntax)

```suggestion
{ lib
, stdenvNoCC
, fetchFromGitHub
, makeWrapper
```

File: pkgs/applications/video/dra-cla/default.nix [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+    owner = "CoolnsX";
+    repo = "dra-cla";
+    rev = "fd5e43bb32b5bc9013382917d1efacda9c3071a8";
+    hash = "sha256-SMtuflVsxe0PWmzabSDy+vhIN2bTqyiYAT/T1ChY+xY=";
+  };
+
+  nativeBuildInputs = [ makeWrapper ];
+
+  installPhase = ''
+      runHook preInstall
```
Comment (figsoda) [inline pkgs/applications/video/dra-cla/default.nix:28]:
```suggestion
    runHook preInstall
```

File: pkgs/applications/video/dra-cla/default.nix [hunk 8]
```diff
@@ -0,0 +1,45 @@
...
+    wrapProgram $out/bin/dra-cla \
+      --prefix PATH : ${lib.makeBinPath [ gnugrep gnused curl mpv aria2 ffmpeg openssl ]}
+
+    runHook postInstall
+  '';
+
+  meta = with lib; {
+    homepage = "https://github.com/CoolnsX/dra-cla";
+    description = "A cli tool to browse and play korean drama, chinese drama";
+    license = licenses.gpl3Plus;
```
Comment (figsoda) [inline pkgs/applications/video/dra-cla/default.nix:41]: Please correct me if I'm wrong, but I couldn't find anything suggesting the `plus` part

```suggestion
    license = licenses.gpl3Only;
```

Comment (idlip) [reply pkgs/applications/video/dra-cla/default.nix:41]: My bad, i had copied template from similar project.

**Comment by idlip** [comment]
@figsoda So sorry on missing on this, this is my first pr, will go through these design principles next time before submitting pr.

Thank you very much for going through!

**Comment by figsoda** [comment]
You don't have to be sorry, nixpkgs is a large project, and it's impossible to go through every bit of information. I make mistakes too, and that's why reviewing is a thing!

**Comment by idlip** [comment]
Thank you for your kind support!

Sure, nix is a giant project, likewise the learning, nonetheless there is always the fun part!

**Review by figsoda — APPROVED** [review]

**Pull request merged**